app/models/proxy_config.rb

```diff
@@ -111,11 +113,14 @@ def clone_to(environment:)
 
   def service_token_exists
     return if service_token
+
     errors.add :service_token, :missing
   end
 
   def api_backend_exists
-    return if proxy&.api_backend_present?
+    # FIXME: we should remove the nil check
+    return if proxy&.api_backend_present? || proxy&.service_mesh_integration?
+
     errors.add :api_backend, :missing
   end
```

Comment:
I am not sure if we should keep a validation called 'api_backend_exists' that adds an error if the backend API does not exist OR it has service mesh integration. It would make more sense to me to either change the name of the method to do as it says it does or change the condition to validate. Something like this:

```ruby
validate :api_backend_exists, on: :create, if: -> { !service_mesh_integration? }

private

delegate :service_mesh_integration?, to: :proxy, allow_nil: true

def api_backend_exists
    # FIXME: we should remove the nil check
    return if proxy&.api_backend_present?

    errors.add :api_backend, :missing
  end
```

Reply:
Cool. Any thoughts about removing the nil check?

Reply:
No thoughts 😄 it is just weird 🙈 for the previous check of proxy not being optional